### PR TITLE
blocks: tag_gate (improvements)

### PR DESCRIPTION
**Pull request by rubund**
- Make it possible to specify a single tag key to look for instead of all tags.
- Extended unit test for new functionality
- Actually fix a bug which made the unit test do nothing earlier

### CODE REVIEW

File: gr-blocks/grc/blocks_tag_gate.xml [hunk 1]
```diff
@@ -54,6 +57,13 @@
       <key>False</key>
     </option>
   </param>
+
```
Comment (mhostetter) [inline gr-blocks/grc/blocks_tag_gate.xml]: Could we also take this opportunity to change the name of the propagate tags parameter from `<name>Propagate_tags</name>` to `<name>Propagate Tags</name>` on line 47? The old name seems kinda wonky.

**Comment by mhostetter** [comment]
I wonder if it's useful to allow multiple (not just single) tag keys to be gated by this block. Or if assumed usage would just be to use multiple Tag Gates, each gating a single key.

**Comment by bastibl** [comment]
I guess since this [commit](https://github.com/gnuradio/gnuradio/commit/080bbf3dfcef968a2141714852bf691ba442bb2f) `setup_rpc` is not called for all blocks

**Comment by rubund** [comment]
@mhostetter I am sure it would be useful. However, I am not so sure how it then should be implemented. Can we provide a list of strings through the swig interface to c++? How would the GRC configuration look like in order to facilitate providing multiple strings? (maybe just have a field which accepts a list)

**Pull request merged**